source/mod_dist.f90

```diff
@@ -98,159 +265,726 @@ subroutine dist_parseInputLine(inLine, iLine, iErr)
+    call chr_cast(inVal, dist_partCol2(partNo), sErr)
+    dist_partCol2(partNo) = dist_partCol2(partNo) * dist_colScale(colNo)
+
+  !  Vertical Coordinates
+  ! ========================
 
-  ! Update longitudinal particle arrays from read momentum
-  call part_updatePartEnergy(2)
+  case(dist_fmtY, dist_fmtYN, dist_fmtJY)
+    call chr_cast(inVal, dist_partCol3(partNo), sErr)
+    dist_partCol3(partNo) = dist_partCol3(partNo) * dist_colScale(colNo)
 
-  if(j < napx) then
-    write(lout,"(a,i0)") "DIST> WARNING Read a number of particles LOWER than requested: ",napx
-    napx = j
+  case(dist_fmtPY, dist_fmtYP, dist_fmtPYP0, dist_fmtPYN, dist_fmtPhiY)
+    call chr_cast(inVal, dist_partCol4(partNo), sErr)
+    dist_partCol4(partNo) = dist_partCol4(partNo) * dist_colScale(colNo)
+
+  !  Longitudinal Coordinates
+  ! ==========================
+
+  case(dist_fmtSIGMA, dist_fmtZETA, dist_fmtDT, dist_fmtZN, dist_fmtJZ)
+    call chr_cast(inVal, dist_partCol5(partNo), sErr)
+    dist_partCol5(partNo) = dist_partCol5(partNo) * dist_colScale(colNo)
+
+  case(dist_fmtE, dist_fmtDEE0, dist_fmtPT)
+    call chr_cast(inVal, dist_partCol6(partNo), sErr)
+    dist_partCol6(partNo) = dist_partCol6(partNo) * dist_colScale(colNo)
+    dist_updtEFrom = 1
+
+  case(dist_fmtP)
+    call chr_cast(inVal, dist_partCol6(partNo), sErr)
+    dist_partCol6(partNo) = dist_partCol6(partNo) * dist_colScale(colNo)
+    dist_updtEFrom = 2
+
+  case(dist_fmtDPP0, dist_fmtPZN, dist_fmtPhiZ)
+    call chr_cast(inVal, dist_partCol6(partNo), sErr)
+    dist_partCol6(partNo) = dist_partCol6(partNo) * dist_colScale(colNo)
+    dist_updtEFrom = 3
+
+  !  Ion Columns
+  ! =============
+
+  case(dist_fmtMASS)
+    call chr_cast(inVal, nucm(partNo), sErr)
+    nucm(partNo)  = nucm(partNo) * dist_colScale(colNo)
+    dist_readMass = .true.
+
+  case(dist_fmtCHARGE)
+    call chr_cast(inVal, nqq(partNo), sErr)
+    dist_readCharge = .true.
+
+  case(dist_fmtIonA)
+    call chr_cast(inVal, naa(partNo), sErr)
+    dist_readIonA = .true.
+
+  case(dist_fmtIonZ)
+    call chr_cast(inVal, nzz(partNo), sErr)
+    dist_readIonZ = .true.
+
+  case(dist_fmtPDGID)
+    call chr_cast(inVal, pdgid(partNo), sErr)
+    dist_readPDGID = .true.
+
+  end select
+
+end subroutine dist_saveParticle
+
+! ================================================================================================ !
+!  Post-Processing of Particle Arrays
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-07-05
+!  Updated: 2019-07-09
+! ================================================================================================ !
+subroutine dist_postprParticles
+
+  use crcoall
+  use mod_common
+  use mod_particles
+  use mod_common_main
+  use physical_constants
+  use numerical_constants
+
+  logical doAction, doNormal
+
+  doAction = .false.
+  doNormal = .false.
+
+  select case(dist_partFmt(6))
+  case(dist_fmtNONE)
+    ejv(1:napx)  = e0
+    call part_updatePartEnergy(1,.false.)
```

Comment:
why not `call part_updatePartEnergy(dist_updtEFrom,.false.)`?

Reply:
dist_updtEFrom will be deleted. After the last rewrite it is redundant.